src/plugins/intel_gpu/src/graph/gemm.cpp

```diff
@@ -108,7 +108,13 @@ std::vector<layout> gemm_inst::calc_output_layouts(gemm_node const& /*node*/, co
 
     std::vector<ShapeType> output_shapes = ov::op::v0::shape_infer(&op, input_shapes);
 
-    return { layout{output_shapes[0], output_type, input0_layout.format, prim->output_paddings[0]} };
+    // Select preferred output format if the impl_type of gemm node is onednn.
+    cldnn::format output_format = input0_layout.format;
+    if (node.get_preferred_impl_type() == impl_types::onednn
```

Comment:
How about use kernel_impl_param instead of node? 
because we are going to avoid using node itself in calc_output_layouts.

```suggestion
    if (impl_param.preferred_impl_type == impl_types::onednn
```

Reply:
You can add  preferred_impl_type and preferred_output_fmt to the kernel_impl_params if necessary.

Reply:
We will change this code when we remove program node from calc_output_layouts.
You can just apply it as is.